icu4c/source/i18n/unum.cpp

```diff
@@ -28,17 +28,83 @@
 #include "unicode/dcfmtsym.h"
 #include "unicode/curramt.h"
 #include "unicode/localpointer.h"
+#include "unicode/measfmt.h"
 #include "unicode/udisplaycontext.h"
 #include "uassert.h"
 #include "cpputils.h"
 #include "cstring.h"
+#include "putilimp.h"
 
 
 U_NAMESPACE_USE
 
 
+// to support UNUM_DURATION
+class MeasureFormatWrapper : public NumberFormat {
+private:
+    Locale locale;
+    MeasureFormat* mf;
+public:
+    MeasureFormatWrapper(const Locale& locale, UErrorCode& err) {
+        this->locale = locale;
+        mf = new MeasureFormat(locale, UMEASFMT_WIDTH_NUMERIC, err);
+        if (mf == nullptr && U_SUCCESS(err)) {
+            err = U_MEMORY_ALLOCATION_ERROR;
+        } else if (U_FAILURE(err)) {
+            delete mf;
+            mf = nullptr;
+        }
+    }
+    
+    ~MeasureFormatWrapper() {
+        delete mf;
+    }
+    
+    NumberFormat *clone() const override {
+        UErrorCode ignoreErr = U_ZERO_ERROR;
+        return new MeasureFormatWrapper(locale, ignoreErr);
+    }
+    
+    UnicodeString &format(double number, UnicodeString &appendTo, FieldPosition &pos) const override {
+        UErrorCode err = U_ZERO_ERROR;
+        if (number < 60) {
```

Comment:
Question: is it documented somewhere that the input is expected to be in seconds, and not minutes, milliseconds, etc?

Reply:
As I recall, RBNF durations are poorly documented. The existing rules always assumed that it was in seconds, and it was only translated into English. When you scale up to beyond English, RBNF durations are too painful to attempt translation.  You have to copy a lot of math around.  For backwards compatibility reasons, I recommend not breaking the English assumptions about how the rules worked.

For reference, it's actually better to use a list format, and to specify what units are in the list.  RBNF isn't well suited to change the base unit. It's a similar translation pattern with miles, feet and inches.